README.md

```diff
@@ -63,37 +63,37 @@ To simplify common tasks there is a Makefile in place. The most common tasks wil
+All components must extend from [\yii\base\Component](http://www.yiiframework.com/doc-2.0/guide-structure-application-components.html) so that they can be configured in the ```components``` section of the application configuration. This also allows them to be accessed via ```\Yii::$app->componentId```. While each component has a defined interface for methods to implement, what properties it needs for configuration are up to each implementation as appropriate. See [our common/config/local.php.dist](https://github.com/silinternational/idp-pw-api/blob/develop/application/common/config/local.php.dist) for examples of configurations.
 
 ### Authentication Component
 We use SAML for authentication but this component can be replaced to support whatever method is needed. For example an auth component could be written to implement OAuth or use Google, etc.
 
 * Component ID: ```auth```
 * Implement interface: ```common\components\auth\AuthnInterface```
-* Example implementation: [idp-pw-api-auth-saml](https://github.com/silinternational/idp-pw-api-auth-saml)
+* [Example implementation](./common/components/auth/Saml.php)
 
 ### Password Store Component
 You can store your passwords wherever you like, whether it is LDAP, Active Directory, a database, or even Redis.
 
 * Component ID: ```passwordstore```
 * Implement interface: ```common\components\passwordStore\PasswordStoreInterface```
-* Example implementation: [idp-pw-api-passwordstore-ldap](https://github.com/silinternational/idp-pw-api-passwordstore-ldap)
+* [Example implementation](./common/components/passwordStore/Ldap.php)
 
 ### Personnel Component
 The personnel component is used to look up informaton about users from your company's personnel system. This includes verifying that they are an active employee, getting information about them like name, email, employee id, whether they have a supervisor and what their supervisors email address is. If the personnel system is aware of spouses it can also provide the spouse's email address.
 
 * Component ID: ```personnel```
 * Implement interface: ```common\components\personnel\PersonnelInterface```
-* Example implementation: [idp-pw-api-personnel-insite](https://github.com/silinternational/idp-pw-api-personnel-insite)
+* [Example implementation](./common/components/personnel/IdBroker.php)
 
 ### Phone Verification Component
 This component is used for performing phone based verification of users.
 
 * Component ID: ```phone```
 * Implement interface: ```common\components\phoneVerification\PhoneVerificationInterface```
-* Example implementation: [idp-pw-api-phoneverification-nexmo](https://github.com/silinternational/idp-pw-api-phoneverification-nexmo)
+* [Example implementation](./common/components/Base.php)
```

Comment:
Is `common/components/Base.php` actually an implementation of "PhoneVerificationInterface"?

Reply:
No, it's not. Thanks for catching it.